web3.md

```diff
@@ -1126,5 +1132,48 @@ Collecting Coverage Data
       requires notBool PCOUNT in PCS
       [priority(25)]
 
+    syntax KItem ::= "#firefly_getCoverageData"
+ // -------------------------------------------
+    rule <k> #firefly_getCoverageData => #sendResponse ("result": #makeCoverageReport(COVERAGE, PGMS)) ... </k>
+         <opcodeCoverage> COVERAGE </opcodeCoverage>
+         <opcodeLists>    PGMS     </opcodeLists>
+
+    syntax JSON ::= #makeCoverageReport ( Map, Map ) [function]
+ // -----------------------------------------------------------
+    rule #makeCoverageReport (COVERAGE, PGMS) => {
+                                                  "coverages": [#coveragePercentages(keys_list(PGMS),COVERAGE,PGMS)],
+                                                  "coveredOpcodes": [#serializeCoverage(keys_list(COVERAGE),COVERAGE)],
+                                                  "programs": [#serializePrograms(keys_list(PGMS),PGMS)] 
+                                                 }
+
+    syntax JSONList ::= #serializeCoverage ( List, Map ) [function]
+ // ---------------------------------------------------------------
+    rule #serializeCoverage (.List, _ ) => .JSONList
+    rule #serializeCoverage ((ListItem({ CODEHASH | EPHASE } #as KEY) KEYS), KEY |-> X:Set COVERAGE:Map ) => { Int2String(CODEHASH):{ Phase2String(EPHASE): [List2JSONList(Set2List(X))] }}, #serializeCoverage(KEYS, COVERAGE)
+
+    syntax JSONList ::= #serializePrograms ( List, Map ) [function]
+ // ---------------------------------------------------------------
+    rule #serializePrograms (.List, _ ) => .JSONList
+    rule #serializePrograms ((ListItem({ CODEHASH | EPHASE } #as KEY) KEYS), KEY |-> X:List PGMS:Map ) => { Int2String(CODEHASH):{ Phase2String(EPHASE): [List2JSONList(X)] }}, #serializePrograms(KEYS, PGMS)
+
+    syntax String ::= Phase2String ( Phase ) [function]
+ // ----------------------------------------------------
+    rule Phase2String (CONSTRUCTOR) => "CONSTRUCTOR"
+    rule Phase2String (RUNTIME)     => "RUNTIME"
+
+    syntax JSONList ::= List2JSONList ( List ) [function]
+ // -----------------------------------------------------
+    rule List2JSONList (.List)                           => .JSONList
+    rule List2JSONList (ListItem(I:Int) L)               => I, List2JSONList(L)
+    rule List2JSONList (ListItem({I:Int | _:OpCode }) L) => I, List2JSONList(L)
```

Comment:
Name this method something more specific (eg. `CoverageIDList2JSONList`), because there are other places we need to convert `List` to a `JSONList` where we don't want it to use this function directly.